frontend/src/layout/header/UserBox.tsx

```diff
@@ -107,149 +171,189 @@ const WithFloatingMenu: React.FC<{ children: ReactElement }> = ({ children }) =>
-                {t("back")}
-            </MenuItem>
-            <LanguageMenu />
+            <ReturnButton onClick={() => close()}>{t("Choose language")}</ReturnButton>
+            <LanguageMenu close={() => close()} />
         </>,
     });
 
-    return (
-        <Floating padding={0}>
+    return <Floating borderWidth={0} padding={0}>
+        <div
+            onClick={e => {
+                if (e.target === e.currentTarget) {
+                    close();
+                }
+            }}
+            onBlur={handleBlur}
+            css={{
+                position: "relative",
+                // Grey out background on mobile devices.
+                [`@media (max-width: ${BREAKPOINT_MEDIUM}px)`]: {
+                    position: "fixed",
+                    top: "var(--header-height)",
+                    bottom: 0,
+                    left: 0,
+                    right: 0,
+                    zIndex: 1001,
+                    backgroundColor: "#000000a0",
+                },
+            }}>
             <ul css={{
-                listStyle: "none",
+                borderRadius: 8,
+                zIndex: 1000,
+                right: 0,
                 margin: 0,
-                paddingLeft: 0,
                 minWidth: 200,
+                paddingLeft: 0,
+                overflow: "hidden",
+                listStyle: "none",
+                [`@media (max-width: ${BREAKPOINT_MEDIUM}px)`]: {
+                    backgroundColor: "white",
+                    marginTop: 0,
+                    position: "fixed",
+                    left: 0,
+                    top: 0,
+                    boxShadow: "none",
+                    borderRadius: "0 0 8px 8px",
+                },
             }}>{items}</ul>
-        </Floating>
-    );
+        </div>
+    </Floating>;
+};
+
+
+const LanguageSettings: React.FC = () => {
+    const { t } = useTranslation();
+
+    return <WithFloatingMenu type="language">
+        <ActionIcon title={t("language")} css={{
+            [`@media not all and (max-width: ${BREAKPOINT_SMALL}px)`]: {
+                button: { padding: 7 },
+            },
+        }}>
+            <HiOutlineTranslate size={24} />
+        </ActionIcon>
+    </WithFloatingMenu>;
+};
+
+
+type ReturnButtonProps = {
+    onClick: () => void;
+    children: ReactNode;
 };
 
+const ReturnButton: React.FC<ReturnButtonProps> = ({ onClick, children }) => (
+    <div css={{
+        borderBottom: "1px solid var(--grey80)",
+        display: "flex",
+        [`@media not all and (max-width: ${BREAKPOINT_MEDIUM}px)`]: {
+            display: "none",
+        },
+    }}>
+        <div onClick={onClick} tabIndex={0} css={{
+            cursor: "pointer",
+            padding: "24px 12px",
+            opacity: 0.75,
+            ":hover, :focus": { opacity: 1 },
+            ":focus": {
+                outline: "none",
+                boxShadow: "inset 0 0 0 2px var(--accent-color)",
```

Comment:
Why inset box shadow instead of outline with offset?

Reply:
Outline with offset was somehow cut off on two sides, but increasing the offset to -2 works, so I changed this now.